### PR TITLE
General Feature Additions

### PR DESCRIPTION
**Pull request by ProdPreva1l**
Implements [Add configurable invulnerability period after teleporting](https://github.com/WiIIiam278/HuskHomes/issues/515)
Implements [Show home privacy status in /homelist](https://github.com/WiIIiam278/HuskHomes/issues/543)

### CODE REVIEW

File: common/src/main/resources/locales/en-gb.yml [hunk 1]
```diff
@@ -64,8 +64,10 @@ locales:
   public_home_list_page_title: '[☀ Public Homes:](#00fb9a) [(%1%-%2% of](#00fb9a) [%3%](#00fb9a bold)[)](#00fb9a)\n'
   warp_list_page_title: '[Warps:](#00fb9a) [(%1%-%2% of](#00fb9a) [%3%](#00fb9a bold)[)](#00fb9a)\n'
   command_list_title: '[HuskHomes](#00fb9a bold) [| Command List](#00fb9a)\n'
-  home_list_item: '[[%1%]](show_text=&#00fb9a&%1%\n&7ℹ %3% run_command=/huskhomes:edithome %2%)'
+  home_list_item: '[[%1%]](show_text=&#00fb9a&%1%\n&7Status: %3%\n&7ℹ %4% run_command=/huskhomes:edithome %2%)'
   public_home_list_item: '[[%1%]](show_text=&#00fb9a&%1%\n&7☻ %3%\n&7ℹ %4% run_command=/huskhomes:phome %2%)'
+  home_is_public: "&#00fb9a&Public"
+  home_is_private: "&#239e6f&Private"
```
Comment (WiIIiam278) [inline common/src/main/resources/locales/en-gb.yml:70]: Can we format these with single quotes for consistency?

File: sponge/src/main/java/net/william278/huskhomes/user/SpongeUser.java [hunk 2]
```diff
@@ -131,4 +131,18 @@ public boolean isVanished() {
...
+     */
+    @Override
+    public void handleInvulnerability() {
+        if (plugin.getSettings().getGeneral().getTeleportInvulnerabilityTime() <= 0) {
+            return;
+        }
+        long invulnerabilityTimeInTicks = 20L * plugin.getSettings().getGeneral().getTeleportInvulnerabilityTime();
+        player.invulnerable().set(true);
+        // Remove the invulnerability
+        plugin.runSyncDelayed(() -> player.invulnerable().set(true), invulnerabilityTimeInTicks);
```
Comment (WiIIiam278) [inline sponge/src/main/java/net/william278/huskhomes/user/SpongeUser.java:145]: typo: needs to be `set(false)` I assume

File: bukkit/src/main/java/net/william278/huskhomes/user/BukkitUser.java [hunk 3]
```diff
@@ -25,10 +25,14 @@
...
 import net.william278.huskhomes.teleport.TeleportationException;
 import net.william278.huskhomes.util.BukkitAdapter;
+import net.william278.huskhomes.util.BukkitTask;
+import net.william278.huskhomes.util.Task;
+import org.bukkit.Bukkit;
 import org.bukkit.entity.Player;
 import org.bukkit.event.player.PlayerTeleportEvent;
 import org.bukkit.metadata.MetadataValue;
 import org.bukkit.permissions.PermissionAttachmentInfo;
+import org.bukkit.plugin.Plugin;
```
Comment (WiIIiam278) [inline bukkit/src/main/java/net/william278/huskhomes/user/BukkitUser.java:35]: Could you auto-cleanup your imports? (If you're using intellij, select the root project and use CTRL+ALT+O)

File: common/src/main/java/net/william278/huskhomes/command/HomeListCommand.java [hunk 4]
```diff
@@ -110,6 +112,8 @@ private Optional<PaginatedList> generateList(@NotNull CommandUser executor, @Not
                         plugin.getLocales()
                                 .getRawLocale("home_list_item",
                                         Locales.escapeText(home.getName()), home.getSafeIdentifier(),
+                                        home.isPublic() ? plugin.getLocales().getRawLocale("home_is_public").orElse("&#00fb9a&Public"):
```
Comment (WiIIiam278) [inline common/src/main/java/net/william278/huskhomes/command/HomeListCommand.java:115]: Let's just supply `Public` or `Private` if no locale is found here :)

File: common/src/main/java/net/william278/huskhomes/user/OnlineUser.java [hunk 5]
```diff
@@ -211,6 +211,12 @@ public Audience getAudience() {
      */
     public abstract boolean isVanished();
 
+    /**
+     * Handles player invulnerability after teleporting
+     *
```
Comment (WiIIiam278) [inline common/src/main/java/net/william278/huskhomes/user/OnlineUser.java:216]: You can probably remove this blank line in the JD comment

**Review by WiIIiam278 — CHANGES_REQUESTED** [review]
Thanks for your work, looks pretty good! :) Appreciate it.

In addition to the review comments, could you propegate your locale changes to the other locale langs?

File: sponge/src/main/java/net/william278/huskhomes/user/SpongeUser.java (hunk 2 above)
Comment (ProdPreva1l) [reply sponge/src/main/java/net/william278/huskhomes/user/SpongeUser.java:145]: whoops!

File: bukkit/src/main/java/net/william278/huskhomes/user/BukkitUser.java (hunk 3 above)
Comment (ProdPreva1l) [reply bukkit/src/main/java/net/william278/huskhomes/user/BukkitUser.java:35]: Yeah i thought i had auto clean up enabled appears not

**Comment by ProdPreva1l** [comment]
Done all changes

**Pull request merged**